lib/puppet/pops/types/types.rb

```diff
@@ -2799,6 +2800,12 @@ def really_instance?(o, guard = nil)
     resolved_type.really_instance?(o, guard)
   end
 
+  # @return `nil` to prevent serialization of the type_expr used when first initializing this instance
+  # @api private
+  def type_expr
+    nil
```

Comment:
Should we consider bypassing `initialize` and instead poke values into instance attributes directly for special cases?